src/context.c

```diff
@@ -293,7 +292,8 @@ static gboolean r_context_configure_target(GError **error)
 	}
 
 	if (context->config->systeminfo_handler) {
```

Comment:
Should we create an empty `GHashTable` if the system info handler is not set? That would avoid extra checks later.

Reply:
Could help. I'll give it a try.